### PR TITLE
Allow route decorators to stack up again

### LINKED ISSUE #1742: Changes to Blueprints application breaks multiply applied routes!!!
Consider the following pattern:

```
@app.get("/url1")
@app.get("/url2")
async def handle_multiple(request):
     return text("some response")
```

The recent change in #1690 breaks that pattern.

It's a common pattern to use multiple `@blueprint.get(...)` and `@app.get(...)` to handle routes that have common behavior with slight variations, like a login process.

The only alternative I can think of is splitting out each of these routes into distinct rump functions that literally forward onto a common function, which inflates from one line declarations to three lines plus the constant common route handler.

### PR DESCRIPTION
**Pull request by ashleysommer**
A PR:https://github.com/huge-success/sanic/pull/1690 broke the ability to stack route decorators.

This PR will allow again route decorators to stack up without causing a function signature inspection crash
Fix #1742

Also adds a test to detect this breakage, and verify its now fixed, and to ensure it stays fixed in the future.

### CODE REVIEW

File: sanic/app.py [hunk 1]
```diff
@@ -194,6 +194,10 @@ def route(
             strict_slashes = self.strict_slashes
 
         def response(handler):
+            if isinstance(handler, (tuple, list)):
```
Comment (yunstanford) [inline sanic/app.py]: Can we add some comments here for explaining?

File: sanic/app.py [hunk 2]
```diff
@@ -205,7 +209,7 @@ def response(handler):
             if stream:
                 handler.is_stream = stream
 
-            routes = self.router.add(
+            new_routes = self.router.add(
```
Comment (yunstanford) [inline sanic/app.py]: new_routes seems a little bit weird.. maybe we can just extend in place.

File: sanic/app.py (hunk 1 above)
Comment (ashleysommer) [reply sanic/app.py]: Added a comment explaining it

File: sanic/app.py (hunk 2 above)
Comment (ashleysommer) [reply sanic/app.py]: Changed to inplace extend.

**Comment by ashleysommer** [comment]
Pushed some changes, added the same fix for `@websocket()` routes too, added a new test for double-stacked websocket routes.

**Pull request merged**